lib/spack/spack/compilers/__init__.py

```diff
@@ -40,6 +40,7 @@
 
 _imported_compilers_module = 'spack.compilers'
 _path_instance_vars = ['cc', 'cxx', 'f77', 'fc']
+_flags_instance_vars = ['cflags', 'cxxflags', 'fflags']
```

Comment:
These should probably include `cppflags` as well, for completeness's sake. I understand why we wouldn't want to add `ldflags` or `ldlibs`.

Are the fortran flags you add in the xl_r compiler absolutely necessary for it to run properly? If not, this should probably be changed in favor of some optional method. We don't want to restrict how spack users configure their compilers any more than necessary.